src/control/server/server.go

```diff
@@ -234,11 +256,32 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 			return err
 		}
 
-		if i == 0 {
+		srv.OnReady(func(ctx context.Context) error {
+			if !mgmtSvc.sysdb.IsLeader() {
```

Comment:
Maybe mgmtSvc should have its own IsLeader() function, rather than exposing the workings of the database here